packages/dataviews/src/dataviews-layouts/grid/index.tsx

```diff
@@ -81,7 +86,27 @@ function GridItem< Item >( {
 				}
 			} }
 		>
-			<div className="dataviews-view-grid__media">
+			<div
+				className={ clsx( 'dataviews-view-grid__media', {
+					'dataviews-view-grid__media--clickable':
+						isItemClickable( item ),
+				} ) }
+				tabIndex={ isItemClickable( item ) ? 0 : undefined }
+				role="button"
+				onClick={ () => {
```

Comment:
this prop should be undefined if the item is not clickable rather than checking within the callback.